syntax = "proto3";

package fixbug;

option cc_generic_services = true;

message ResultCode {
  int32 errcode = 1;
  bytes errmsg = 2;
}

message LoginRequest {
  bytes name = 1;
  bytes pwd = 2;
  map<int32, string> test = 3;
}

message LoginResponse {
  ResultCode result = 1;
  bool success = 2;
}

message GetFriendListsRequest {
  uint32 userid = 1;
}

message User {
  bytes name = 1;
  uint32 age = 2;
  enum Sex {
    MAN = 0;
    WOMAN = 1;
  }
  Sex sex = 3;
}

message GetFriendListsResponese {
  ResultCode result = 1;
  repeated User friend_lists = 2;
}

service UserServiceRpc {
  rpc Login(LoginRequest) returns(LoginResponse);
  rpc GetFriendLists(GetFriendListsRequest) returns(GetFriendListsResponese);
}